### PR TITLE
[browser] cleanup before memory snapshot

### PR DESCRIPTION
**Pull request by pavelsavara**
Split from https://github.com/dotnet/runtime/pull/82049

- moved get_preferred_icu_asset
- moved TZ detection
- moved runtimeHelpers.waitForDebugger
- replaced `anyModule` with `EmscriptenModuleInternal` type
- added runtimeHelpers.updateGlobalBufferAndViews
- removed obsolete comments about blazor startup sequence
- removed duplicate calls mono_wasm_init_diagnostics (also previously blazor startup path)

### CODE REVIEW

**Review by maraf — APPROVED** [review]

File: src/mono/wasm/runtime/imports.ts [hunk 1]
```diff
@@ -10,6 +10,7 @@ import type { EmscriptenModule } from "./types/emscripten";
 
 // these are our public API (except internal)
 export let Module: EmscriptenModule & DotnetModule;
+export let anyModule: any = undefined;
```
Comment (kg) [inline src/mono/wasm/runtime/imports.ts:13]: Can we explain what `anyModule` is/means, somewhere? I don't even see references to it in the diff

**Review by ilonatommy — APPROVED** [review]

File: src/mono/wasm/runtime/imports.ts (hunk 1 above)
Comment (pavelsavara) [reply src/mono/wasm/runtime/imports.ts]: I introduced new internal type `EmscriptenModuleInternal` instead

**Review by lambdageek — APPROVED** [review]
threads and diagnostics looks ok

**Comment by pavelsavara** [comment]
CI failure `undefined symbol: mono_wasm_pthread_on_pthread_attached` is fixed here https://github.com/dotnet/runtime/pull/82826
CI failure ` System.IO.IOException: 'NuGet-Migrations'` is https://github.com/dotnet/runtime/issues/80619

**Pull request merged**